shared/decode.ts

```diff
@@ -0,0 +1,26 @@
+import { decodeAbi } from "@spec.dev/core";
+
+export function decodeRfpRegistrationData(useRegistryAnchor: boolean, data: any) {
+  if (useRegistryAnchor) {
+    const [recipientId, proposalBid, metadata] = decodeAbi(data, [
+      "address",
+      "uint256",
+      "tuple(uint256, string)",
+    ]);
+
+    return {
+      proposalBid,
+      metadata,
+    };
+  } else {
+    const [recipientAddress, registryAnchor, proposalBid, metadata] = decodeAbi(
+      data,
+      ["address", "address", "uint256", "tuple(uint256, string)"]
+    );
+
+    return {
+      proposalBid,
+      metadata,
+    };
+  }
+}
```

Comment:
Adding helper function right below here to reformat the metadata array values back into struct format for callers to use. Unfortunately when using the `decodeAbi` function to decode a tuple, all it gives you back are the array values. And `uint256` types are returned as `BigInt` types, so lets swap that back to just a number since we know the protocol id will be a small number.

```typescript
function formatMetadataAsStruct(metadataValues: any[]): { [key: string]: any } {
    const [protocol, pointer] = metadataValues || []
    return { protocol: protocol?.toNumber(), pointer }
}
```